ClipboardAPI/clipboard-change-event-explainer.md

```diff
@@ -0,0 +1,209 @@
+
+```html
+<iframe
+    src="https://other-domain.com/will-monitor-clipboard.html"
+    allow="clipboard-read;"
+>
+</iframe>
+```
+
+This event is not accessible within a "fencedframe" since the clipboard and the clipboardchange event could be used as a communication channel between the host and the fencedframe, so constitute a privacy threat.
+
+#### 5.1.2 Approach 2 - No permission required
+Since no data is being sent as part of the clipboardchange event, it can be argued that we don't need any permission to simply know when clipboard contents change. This will simplify the user flow as they don't need to explicitly ask for permissions before listening to the event. It can also be noted that there is no identified incremental security risk in providing the "clipboardchange" event to web authors since this event doesn't allow the site to modify the system in any way - it simply tells when the clipboard contents have changed.
+
+##### Pros
+1. Simpler implementation and user experience
+
+##### Cons
+1. This can cause user privacy violation - a site will be able to monitor the clipboard - a privileged OS component - without the user explicitly allowing it. (Though the author would only know that the clipboard changed, but wouldn't have any details about the payload)
+
+#### 5.1.3 Conclusion
+We favour Approach 1 i.e. having clipboard-read permission required to listen to clipboardchange event, because it has more provisions which safeguards user privacy.
+
+### 5.2 Page focus requirement
+As per the [current spec](https://www.w3.org/TR/clipboard-apis/#clipboard-event-clipboardchange), we should not fire "clipboardchange" event when a page is not is focus. This is in-line with the current behavior where async clipboard API is not accessible unless the given page is in focus. We do fire "clipboardchange" event when the page regains focus, incase the clipboard contents had changed when the page was out of focus. Note that even if the clipboard had changed multiple times while the page was out of focus, we will only fire a single "clipboardchange" event when the page regains focus. This is because the event is designed to indicate that the clipboard contents are different from what they were when the page lost focus, rather than tracking every individual change that occurred while the page was out of focus.
+
+#### 5.2.1 Approach 1 (Preferred) - Page required to be in focus to receive event
+
+##### Pros
+1. This is in-line with current async clipboard focus APIs which require focus to access.
+2. Can simplify implementation since browsers can simply check for clipboard change on page focus. However, this needs to be further investigated.
```

Comment:
"However, this needs to be further investigated." -> We should elaborate on what is meant here or simply exclude this statement.